architecture/2024-10-user-defined-types-authoring-experience-feature-spec.md

```diff
@@ -0,0 +1,230 @@
+                    type: 'string'
+                    description: 'Name of the queue'       
+                port:
+                    type: 'string'
+                    description: 'Port'
+                connectionString:
+                    type: 'string'
+                    description: 'Connection string to the messaging service'
+                logAnalytics:
+                    type: 'string'
+                    description: 'Usage type of the messaging service'  
+                    default: 'off'        
+```
+
+Deb creates the updated resource type schema via the CLI
+
+```bash
+rad resource-type create -f plaid.yaml
+Resource type `plaid` created successfully for version `2024-11-01`
+```
+
+Question 
+
+1. We leave it up to the user to decide whether to use the new optional property in their application definitions and Recipes. Should we provide any guidance?
+2. If Deb tries to add an optional property within the same version or without a default value, what should be the behavior ? Should we consider that any change warrants a api version upgrade like Azure/ ARM does?
+
+#### Sub-scenario 2: Deb adds a required property and updates the resource type Plaid.
+Deb wants to add a required property `messageForwarding`. This is considered to be a breaking change as it would require the Infrastructure operator teams and developers to update their Recipes and application definitions to use the new property in the resource type Plaid. Deb should be provided with an error message when a breaking change is detected in the schema definition.
+
+```yaml
+namespace: 'Mycompany.Messaging'
+resourceTypes:
+    'Mycompany.Messaging/plaidResource':
+        name: 'Name of the resource'
+        version: '2024-11-01'
+            properties: 
+                required: ['host', 'port', 'connectionString']
+                queueName:
+                    type: 'string'
+                    description: 'Name of the queue'   
+                host:
+                    type: 'string'
+                    description: 'Hostname of the messaging service'        
+                port:
+                    type: 'string'
+                    description: 'Port'
+                connectionString:
+                    type: 'string'
+                    description: 'Connection string to the messaging service'
+                messageForwarding:
+                    type: 'string'
+                    description: 'Usage type of the messaging service'          
+```
+
+Deb creates the updated resource type schema via the CLI
+
+```bash
+rad resource-type create -f plaid.yaml
+Error: Unable to create the resource-type `plaid` for version `2024-10-01`. The schema definitions has a breaking change with the required property `instance_type`. Please delete older versions of the resource type and try again.
+```
+
+Questions
+1. For MVP, can we just flag breaking changes and provide documentation on how to handle breaking changes ? 
```

Comment:
Either breaking changes are explicitly supported, or explicitly blocked. If they are supported, we have to define what the behavior is. 

I'd lean towards explicitly blocked to start, and we can add ways to make them explicitly supported in the future.